### DISCUSSION TITLE
Deployment guide

### DISCUSSION THREAD

**Discussion by tablufc88** [saleor/saleor#9088] [Deployment]
Just wondering if anybody has an up to date step by step deployment guide? After a few days testing & loving the saleor-platform, I'm struggling to get Saleor core running in production with Docker or manual install. Would prefer manual install as we don't use Docker but any help appreciated.

**Comment by karolkielecki** [comment]
Hi @tablufc88, 
Saleor-platform repo hasn't been designed for production as stated in the readme section. Docker is a preferred method, but some of our community members use e.g., Heroku. For Dashboard or Storefront (static apps), we recommend using e.g., S3.

**Reply by tablufc88** [reply]
Hey @karolkielecki apologies I didn't go into further detail - I had read that on the saleor-platform readme & had also gone through all similar issues & discussions before posting.

What I was looking for was guidance to help deploy the Saleor Django app without Docker (only because we've never used it). Already got the static sites running ok. I had also been following some [steps](https://github.com/neerajgupta2407/saleor_course_resources/blob/master/setup_saleor_core.txt) but they weren't working for me.

Figured it out though. There appears to be a few issues with the above to use the latest version:

- Use python 3.9 instead of python 3.7
- Use python3.9-venv instead of virtualenv
- Install wheel before requirements.txt
- Update to pg13

**Reply by lack-of-gravitas** [reply]
What I'd love is a clear and simple guide to deploy to production, Docker or otherwise and a better explanation of ENV vars.... have spent several days looking and none of the materials available are sufficient or adequate.

**Reply by karolkielecki** [reply]
Hi @lack-of-gravitas, do you have any preferred deployment method? Is our documentation not sufficient, e.g., regarding deploying Saleor on Heroku? https://docs.saleor.io/docs/3.x/developer/running-saleor/heroku

There is an [old Spectrum thread](https://spectrum.chat/saleor/development/deploying-saleor-api-with-docker-and-ssl~74a2f215-22c1-4aed-8273-f6286817f6a6) where one of our community members explains more in case of Docker deployment. 

Btw, what would you like to achieve? If you're new to Saleor, the easiest way to start working with Saleor is to use Saleor CLI-- you can create a free developer account, play with sandboxes, and build a storefront. More info here: https://docs.saleor.io/docs/3.x/cli

**Reply by lack-of-gravitas** [reply]
Our client's preferred route is docker on ACIs or fly.io but I'll see if I can deploy a Prod version to Heroku. I can pull and run the multi-container docker saleor-platform locally but Prod has eluded me. 

At this stage I'll take anything -- there's a few issues that would confuse any new developer
-- dockerhub registry out of date, have to go trawling through GitHub issues to find that ghcr.io registry should be used
-- unclear use / application of docker-compose / Dockerfiles and path to production
-- possibly missing (maybe too obvious) instructions on application of ENV vars esp. the dreaded "ALLOWED_HOSTS" -- I can see from historical issues that this is a common bug bear and I still haven't grok'd how the three saleor components stitch together using env variables
-- using https://docs.saleor.io/docs/3.x/developer/running-saleor/docker doesn't align to the saleor-platform repo.. Seems like it references saleor [sic core] repo. Also contain any references to updating / setting Prod env variables. Is this assumed knowledge?
-- Azure Storage is supposedly supported due to closed GitHub issues but documentation is not updated (and doesnt appear to be on a PR to update)


examples of past issues whose resolution should be documented permanently --
https://github.com/saleor/saleor/issues/6580
https://github.com/saleor/saleor/issues/9680
https://github.com/saleor/saleor/pull/9866
